File: commands/commit-commands/status.js
import simpleGit from "simple-git";

const git = simpleGit();

export default {
  command: "status",
  describe: "Show the status of the repository",
  builder: () => {},
  handler: async () => {
    try {
      const status = await git.status();
      console.log("Repository status:");
      console.log(
        status.isClean() === true ? "Clean" : "Modified files:",
        status.modified
      );
    } catch (error) {
      console.error("Error retrieving repository status:", error.message);
    }
  },
};

File: cmd/status.go
package cmd

import (
	"os"
	"os/exec"

	"github.com/spf13/cobra"
)

var StatusCmd = &cobra.Command{
	Use:   "status",
	Short: "Show status of your local repository",
	Run: func(cmd *cobra.Command, args []string) {
		if err := gitStatus(); err != nil {
			cmd.Println("Error executing 'status' command: ", err)
			return
		}

		cmd.Println("Repository is clean")
	},
}

func gitStatus() error {
	cmd := exec.Command("git", "status")
	cmd.Stdout = os.Stdout
	cmd.Stderr = os.Stderr
	return cmd.Run()
}